composer.json

```diff
@@ -10,12 +10,11 @@
     ],
     "minimum-stability": "dev",
     "require": {
-        "php": ">=5.3.0",
-        "art4/json-api-client": "0.9.*"
+        "php": ">=5.3.0"
     },
     "autoload": {
-        "psr-0": {
-            "Patreon": "src/"
+        "psr-4": {
+            "Patreon\\": "src/"
```

Comment:
Can you help me understand what these lines are doing? And what is psr-4 vs psr-0

Reply:
These lines define the namespace for our package. Ie:

Say, this is the codebard vendor. (using the dev example) And the project is patreon-php project. When the package is installed via composer, it is put in:

vendor/codebard/patreon-php.

Then, these lines in composer define the namespace. 

So, Codebard\\ is set to the src/ directory in the last 2 lines (psr4 one). Which means that if any class is requested from Codebard namespace, it will be searched for in vendor/codebard/patreon-php (this project's folder) + src.

So the precise search location then becomes vendor/codebard/patreon-php/src.

Then when any class which has that namespace defined is requested, it can be just requested within that namespace like...

use Codebard\OAuth;
$oauth_client = new OAuth($client_id, $client_secret);

...without needing to deal with directory structure, includes etc.

The difference of psr-0 vs psr-4 is, 4 is the later standard and it makes the namespace definitions easier and prevents some directory complication as depicted in the article below. ( after the first section )

https://www.sitepoint.com/battle-autoloaders-psr-0-vs-psr-4/

Ie:

vendor/vendor_name/package_name/src/Vendor_Name/Package_Name/ClassName.php 

...with psr-0 vs 

vendor/vendor_name/package_name/src/ClassName.php 

...with psr-4

So it simplifies the dir structure (one of our issues in earlier one i believe) and it is the latest standard.

Reply:
👍